### PR TITLE
Box control units: Ensure custom units are preserved.

### PR DESCRIPTION
**Pull request by scruffian**
## Description
This is broken out of https://github.com/WordPress/gutenberg/pull/27579 since that PR is blocked.

## How has this been tested?
- Set custom padding on a block using a non-px value
- Ensure that when the box control loads it has the correct units

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/group/block.json [hunk 1]
```diff
@@ -24,6 +24,9 @@
 		},
 		"spacing": {
 			"padding": true
+		},
+		"__experimentalBorder": {
+			"radius": true
```
Comment (youknowriad) [inline packages/block-library/src/group/block.json:29]: This change is unrelated right?

Comment (scruffian) [reply packages/block-library/src/group/block.json:29]: Oops yeah sorry, let me remove that

**Review by youknowriad — APPROVED** [review]
LGTM 👍
